mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -293,34 +293,33 @@ case class ScalaPresentationCompiler(
       val sourceFile = CompilerInterfaces.toSource(params.uri, params.text)
       driver.run(uri, sourceFile)
 
-      val ctx = driver.currentCtx
+      given ctx: Context = driver.currentCtx
       val pos = sourcePosition(driver, params, uri)
       val trees = driver.openedTrees(uri)
-      val path = Interactive.pathTo(trees, pos)(using ctx)
-      val tp = Interactive.enclosingType(trees, pos)(using ctx)
-      val tpw = tp.widenTermRefExpr(using ctx)
+      val path = Interactive.pathTo(trees, pos)
+      val tp = Interactive.enclosingType(trees, pos)
+      val tpw = tp.widenTermRefExpr
 
-      if (tp.isError(using ctx) || tpw == NoType || tpw.isError(using ctx))
+      if (tp.isError || tpw == NoType || tpw.isError)
         ju.Optional.empty()
       else {
-        Interactive.enclosingSourceSymbols(path, pos)(using ctx) match {
+        Interactive.enclosingSourceSymbols(path, pos) match {
           case Nil =>
             ju.Optional.empty()
           case symbols =>
-            val printer = SymbolPrinter()(using ctx)
+            val printer = SymbolPrinter()
             val docComments =
-              symbols.flatMap(ParsedComment.docOf(_)(using ctx))
-            val keywordName = symbols.headOption.map { symbol =>
-              printer.fullDefinition(
-                symbol,
-                tpw
-              )
-            }
-            val typeString = symbols.headOption.map { symbol =>
+              symbols.flatMap(ParsedComment.docOf(_))
+            val hoverString = symbols.headOption.map { symbol =>
```

Comment:
Improved it to return the full hover string here, full separation was actually problematic to maintain.